### PR TITLE
Re-invoke DismissIntent in Autocomplete if ignored

### PR DESCRIPTION
**Pull request by markusaksli-nc**
Ensures the `DismissIntent` used to dismiss the `RawAutocomplete` menu is re-invoked if the menu isn't visible.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by justinmc — COMMENTED** [review]
I think it should be possible to test by wrapping Autocomplete in Actions and hitting escape when the options are not visible.  Let me know if not when you get back to this though.  Otherwise this PR looks like exactly what I was expecting, thanks for opening it so quick.

**Comment by markusaksli-nc** [comment]
@justinmc that did the trick, made sure it kills the mutant as well.

File: packages/flutter/test/widgets/autocomplete_test.dart [hunk 1]
```diff
@@ -883,6 +883,65 @@ void main() {
...
+      ),
+    );
+
+    // Enter text to show options.
+    focusNode.requestFocus();
+    await tester.enterText(find.byKey(fieldKey), 'ele');
+    await tester.pumpAndSettle();
+    expect(find.byKey(fieldKey), findsOneWidget);
+    expect(find.byKey(optionsKey), findsOneWidget);
+
```
Comment (justinmc) [inline packages/flutter/test/widgets/autocomplete_test.dart:932]: Nit: What if you added one more step here for completeness:  Long press the text and assert that the text selection toolbar shows up.  Press escape and assert that it goes away, the options are still visible, and wrappingActionInvoked is false.  Then proceed with the rest of the test.

That should be how it works, right?

**Review by justinmc — APPROVED** [review]
LGTM 👍 

One nit to make the test more comprehensive, assuming it works like I'm thinking it does.  Up to you if you want to add it or not, though.

File: packages/flutter/test/widgets/autocomplete_test.dart (hunk 1 above)
Comment (markusaksli-nc) [reply packages/flutter/test/widgets/autocomplete_test.dart:932]: Doesn't seem within the scope of this test for me. This fix doesn't really affect the text selection toolbar dismiss behavior in an `Autocomplete` so the added asserts wouldn't fail either way.

**Pull request merged**